src/platform/plugins/shared/discover/public/application/main/components/layout/cascaded_documents/blocks/cascade_leaf_component.tsx

```diff
@@ -0,0 +1,304 @@
+          </div>
+        </div>
+      </div>
+      <React.Fragment>{footerRow}</React.Fragment>
+    </div>
+  );
+});
+
+export const ESQLDataCascadeLeafCell = React.memo(
+  ({
+    cellData,
+    cellId,
+    dataGridDensityState,
+    showTimeCol,
+    dataView,
+    services,
+    showKeyboardShortcuts,
+    renderDocumentView,
+    externalCustomRenderers,
+    getScrollElement,
+    getScrollMargin,
+    getScrollOffset,
+  }: ESQLDataCascadeLeafCellProps) => {
+    const [sampleSize, setSampleSize] = useState(cellData.length);
+    const [expandedDoc, setExpandedDoc] = useState<DataTableRecord | undefined>();
+    const [cascadeDataGridDensityState, setCascadeDataGridDensityState] = useState<DataGridDensity>(
+      dataGridDensityState ?? DataGridDensity.COMPACT
+    );
+
+    // TODO: Implement column selection logic,
+    // probably requires a new selection component that will be used within the row
+    const [selectedColumns, setSelectedColumns] = useState<string[]>([]);
+
+    const [isCellInFullScreenMode, setIsCellInFullScreenMode] = useState(false);
+
+    const renderCustomToolbarWithElements = useMemo(
+      () =>
+        getRenderCustomToolbarWithElements({
+          leftSide: (
+            <React.Fragment>
+              <EuiText size="s">
+                <b>
+                  <FormattedMessage
+                    id="discover.esql_data_cascade.row.cell.toolbar.heading"
+                    defaultMessage="{count, plural, =0 {no results} =1 {1 result} other {# results}}"
+                    values={{ count: cellData.length }}
+                  />
+                </b>
+              </EuiText>
+            </React.Fragment>
+          ),
+        }),
+      [cellData]
+    );
+
+    const setExpandedDocFn = useCallback(
+      (...args: Parameters<NonNullable<UnifiedDataTableProps['setExpandedDoc']>>) =>
+        setExpandedDoc(args[0]),
+      [setExpandedDoc]
+    );
+
+    const renderCustomCascadeGridBodyCallback = useCallback(
+      ({
+        Cell,
+        visibleColumns,
+        visibleRowData,
+        setCustomGridBodyProps,
+        gridWidth,
+        headerRow,
+        footerRow,
+      }: EuiDataGridCustomBodyProps) => (
+        <CustomCascadeGridBodyMemoized
+          key={isCellInFullScreenMode ? `full-screen-${cellId}` : cellId}
+          Cell={Cell}
+          data={cellData}
+          visibleColumns={visibleColumns}
+          visibleRowData={visibleRowData}
+          headerRow={headerRow}
+          footerRow={footerRow}
+          gridWidth={gridWidth}
+          getScrollMargin={getScrollMargin}
+          setCustomGridBodyProps={setCustomGridBodyProps}
+          getScrollElement={getScrollElement}
+          initialOffset={getScrollOffset}
+          isFullScreenMode={isCellInFullScreenMode}
+        />
+      ),
+      [cellData, cellId, getScrollElement, getScrollMargin, getScrollOffset, isCellInFullScreenMode]
+    );
+
+    return (
+      <EuiPanel paddingSize="none">
```

Comment:
Some layout issues when do results can be shown

<img width="1336" height="293" alt="Screenshot 2025-11-10 at 11 47 43" src="https://github.com/user-attachments/assets/32939eba-87df-4f97-bf1e-6809e0ccc5f0" />